Fix bfl + revolver firing and reloading issues

Fueredoriku opened the pull request:
- Fix lazur not consuming ammo due to too late animation event trigger
- Fix lazur shooting an extra beam with revolver while reloading

toberge commented on Assets/LazurFiringAnimator.cs:
```diff
@@ -28,6 +30,8 @@ public void EndFiring()
 
     public override void OnFire(GunStats stats)
     {
+        if (IsDiabledInReload && stats.Ammo <= 1)
```
This should actually be in all the firinganimators I think (at least for coil animator as well since it also is possible to start firing that one during reload)

Fueredoriku replied:
It could, but I'm very worried about unforseen regressions on some of the implementations. I'll add it to the coil manually for now and I'll put up an refactor issue we can do after the update so we have the time properly test before this is released 👍

toberge reviewed (approved):
Fiksen funker, og bør også brukes for andre våpendeler.

Ideelt sett burde vi dog få tilbake isFiring som jeg begynte med i tidligere forsøk på å fikse dette problemet, men det er betydelig mer kinkig.

Fueredoriku committed:
Fix bfl/coil + revolver firing and reloading issues

- Fix lazur not consuming ammo due to too late animation event trigger
- Fix lazur shooting an extra beam with revolver while reloading
